frame/babe/src/lib.rs

```diff
@@ -816,6 +849,36 @@ impl<T: Config> Pallet<T> {
 		this_randomness
 	}
 
+	/// Returns the session index that was live when the given epoch happened,
+	/// taking into account any skipped epochs.
+	///
+	/// This function is only well defined for epochs that actually existed,
+	/// e.g. if we skipped from epoch 10 to 20 then a call for epoch 15 (which
+	/// didn't exist) will return an incorrect session index.
```

Comment:
wouldn't it be better to return an error?

Reply:
The problem is we'd need to keep track of extra information in order to detect this. I'll see if there's an easy way to do it.